tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
+aforementioned ordering (which induces a natural partition of the weight 
+shares) to each validator that made an announcement. This is included in the
+block. Once this block is on the blockchain, the next step of the DKG can proceed.
+
+## Dealers
+In a single DKG instance, a subset of validators holding at least
+\\( \frac{2}{3} \\) of the weight shares must each act as dealers. It
+is the responsibility of a dealer to post a signed message to the blockchain
+which includes the session id (\\( \tau \\)) as well the data of the
+PVSS (we do not concern ourselves with the details here). The creation of this
+message is handled by the Ferveo library.
+
+This means that the signed message is a transaction that must be gossiped and
+eventually included on the blockchain. For this particular transaction, there
+is a special verification that must be done, otherwise the transaction should
+not be included in any proposed block. This verification is provided by the
+Ferveo library.
+
+## Aggregation
+
+At some point, enough validators (meaning their combined weight  shares
+exceeds \\( \frac{2}{3}\\) of the total) have their PVSS on the chain 
+for the session with id \\( \tau \\). At this point, the next block proposer can
+aggregate the messages into a single instance using Ferveo. This is then 
+included in the next block. Once this DKG is on the blockchain, it can be 
+used for encryption of transactions until the next DKG instance is run.
+
+When deciding to aggregate the PVSS instances into a single one, we shall 
+consider the instances ranked in terms of the weights of their dealers (in 
+descending order). The topmost dealers constituting enough weight are then 
+chosen and aggregated in their specified order.
+
+We note that enough weight may be gossiped to the current proposer and thus
+they can put the PVSS transactions and aggregation all in the same proposed 
+block.
+
+## Notes on pipelining
+
+Note that in the decription above, at least two blocks must be submitted to the blockchain in order to
+produce the next public encryption key. The first block determines the 
+partition of weight shares of the validators and the last aggregates the 
+PVSS instances from the dealers leading to the final key.
+
+In practice, it may take even longer as the protocol waits for enough PVSS 
+instances to get onto the blockchain. Depending on the chosen refresh rate
+of the keys, another instance of the DKG protocol may need to be started 
+before the current one finishes. This is possible so long as all data
+posted for a DKG instance includes the session id \\( \tau \\).
+
+This means that the storage of a node must be adapted. Each instance of a
+DKG protocol from Ferveo is a state machine that keeps all necessary data
+from the protocol instance. We thus need to store the following data
+ - A list of sesssion ids from the last completed DKG instance up until the
+   latest started (we may prune ids for failed instances however) 
```

Comment:
for simplicity, maybe it's easiest to use an epoch number (or something easily derivable from the block heights) as the session id?

Reply:
I was thinking something like semver where we do E.I where E is the epoch and I is an instance number starting at 0 for each epoch and incremented as necessary if we have to do multiple runs in an epoch due to failures and whatnot.